### ISSUE TITLE
Map type constants fail in queries

### ISSUE THREAD

**Issue by alex-zaitsev** [ClickHouse/ClickHouse#29081]
This query fails with an Exception:

```
with cast([('a', 1)] as Map(String, Int32)) as m select m[name] from system.parts limit 1;

Received exception from server (version 21.8.5):
Code: 44. DB::Exception: Received from taxi9.tenant-a.staging.altinity.cloud:9440. DB::Exception: Illegal column Const(Map(String, Int32)) of first argument of function arrayElement. 
```

It works if m[name] is replaced with materialize(m)[name]

**Comment by nvartolomei** [comment]
Seems fine in today's master build.

**Comment by CurtizJ** [comment]
Fixed in #28096.

**Comment by larry-cdn77** [comment]
Confirmed issue present in version 21.8 LTS
